code/game/turfs/turf.dm

```diff
@@ -14,6 +14,13 @@
 	var/to_be_destroyed = 0 //Used for fire, if a melting temperature was reached, it will be destroyed
 	var/max_fire_temperature_sustained = 0 //The max temperature of the fire which it was subjected to
 
+	var/turf_integrity = 100
```

Comment:
This needs to be on atom level I don't really like having multiple integrity systems that would mostly be copy paste. This is not good for OOP.

Reply:
I know but the move to atom level could be done in the future

Reply:
No it shouldnt, that should be done first in a separate PR if anything. Refactors before reworks. We have no way of knowing you will make atom/integrity after this PR and there's no benefit to the code quality to have this merged before that. only disadvantages because now we have split-up integrity which means more code to clean.

Reply:
i've replaced the integrity with heat_resistance vars and made it simpler